### DISCUSSION TITLE
Powerwall Degradation

### DISCUSSION THREAD

**Discussion by jasonacox** [jasonacox/Powerwall-Dashboard#264] [General]
How are you Powerwall battery capacities holding up over time?

The graph of my two powerwalls since I started measuring:

<img width="1161" alt="image" src="https://user-images.githubusercontent.com/836718/235579788-8c7d8664-778c-4809-b8c6-7e9a885abaf1.png">

I want to add a plot line on there for the "expected degradation" to see how I'm trending. The warranty states that it should have at least 70% of the original capacity remaining at ten years. Anyone know how to add that?

**Comment by jasonacox** [comment]
Also - saw this great question on [Reddit](https://www.reddit.com/r/Powerwall/comments/134ct03/powerwall_degradation_and_charging_efficiency/?utm_source=share&utm_medium=web2x&context=3): Does the charging efficiency stay (relatively) constant no matter the maximum capacity, or is the efficiency reduced when the maximum capacity is reduced?

It would be interesting to see if we could determine and graph charing efficiency (e.g. $\Delta level/t$ per $kW$).

**Reply by jgleigh** [reply]
Do you have a Powerwall+ and a Powerwall2? I've got two of each. How do the 1-4 correspond to the actual units?

<img width="1964" alt="Screenshot 2023-05-03 at 6 43 41 PM" src="https://user-images.githubusercontent.com/654130/236091020-2a4dfbe5-dac0-42e1-a1dd-ec7bf0cecbed.png">

**Reply by jasonacox** [reply]
Yes, I do!  You can see what telegraf is pulling by looking at: http://localhost:8675/pod

In your graph, it does look like 1 & 2 swapped with 3 & 4 last July?  The pypowerwall code assembles and assigns those PW numbers in sequence from the http://localhost:8675/vitals (protobuf) payload.  It assumes the protobuf sequence is always the same, which in general it is, but now your example has me wondering.

You can see the code here:
https://github.com/jasonacox/pypowerwall/blob/c9f20d7c20f99aae102fea9b70c4549109cf1e56/proxy/server.py#L218-L241

**Reply by jgleigh** [reply]
1&2 (at least now) are the Powerwall2 and 3&4 are the Powerwall+. Probalby no way to know how they reported last June.

**Comment by zip90** [comment]
Hello 
I finally settled on a raspberry pi 4 with the rasp OS to run the dashboard. I am not a soft ware programmer. please let me know how to work out the PW 2 capacity degradation overtime from the dashboard. I like to thank to all those who have made it possible to get the PW function information to the end user.

**Reply by jasonacox** [reply]
Hi @zip90  - congrats on the new RPi 4 host!  Do you have the dashboard running?  If so, scroll down to the "Powerwall Capacity" panel and click on the PW1 legend and at the top, select duration dropdown and pick the time period you want to see (e.g. Last 2 Years).  The capacity panel will show the max capacity of the Powerwall over time.  If you have more than one PW, you can Shift-Click PW2, PW3, etc. to show those at the same time.

<img width="207" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/e09a9eaa-849d-4cf4-95f2-606a88bc9e91">

<img width="964" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/a44fe994-7bae-410c-824b-20d092f7644b">

**Reply by jgleigh** [reply]
@jasonacox I've been meaning to ask, what's the difference between PWx and PWx Charge?

**Reply by jasonacox** [reply]
Hi @jgleigh  - 
* The "PWx Charge" line is the current charge - energy remaining (`POD_nom_energy_remaining`) which varies as you charge/discharge the Powerwall.
* The "PWx" line shows the full charge capacity (`POD_nom_full_pack_energy`) which is what we see degrade over time.

You can see the "charge" cycles for my system here with the "capacity" line at the top.

<img width="1081" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/be4e4c4f-18ce-450f-922d-6aa83df37993">

**Reply by zip90** [reply]
@jasonacox 
![IMG_1029](https://github.com/jasonacox/Powerwall-Dashboard/assets/108314347/fe4d0972-4d02-4498-8453-5dacb705ad97)
Thanks for the quick reply ,i have uploaded the PW capacity and charge. To monitor degradation over time do i follow the PW charge and see if it drops below 70% of 15.1Kwh . How come the PW1 in your image is 13.5Kwh

**Reply by jasonacox** [reply]
Use "PW1" not "PW1 Charge" to see the degradation.  It will show the max "capacity" which is what degrades over time.

**Reply by zip90** [reply]
thank you

**Comment by zip90** [comment]
PW capacity monitoring what is Max and min. In my thinking the KWH will be
gradually reducing  over time ,so what is the min. Is it  that the Max is
the day one charge and over time when the capacity reduce it  is visible as
Min , thanks for the reply

**Reply by jasonacox** [reply]
The Powerwalls have an algorithm to instantaneously compute their "fully charged" capacity in kWh (think of it as the size of the fuel tank). That is what the dashboard is pulling for the "PW1" values.  The Powerwalls also have a way to report "current charge" which is the amount of kWh (think of it as how much fuel is in the tank).

* PW1 - Max is the maximum "fully charge" capacity (size of tank) that the Powerwall reported since you have been measuring (Min is the minimum).  The capacity will vary (up and down) but general trend is down over time (degradation)
* PW1 Charge - Max is the maximum "current charge" (how full the tank got) and will cycle as you charge/discharge over time.  Min is the minimum amount.

**Reply by zip90** [reply]
Thank you

**Comment by ThePnuts** [comment]
I was installed April 2022, but didn't get grid approval until Sept so I used the battery heavily in this time to maximize the solar since I couldn't send to grid.

Once I got grid approval, the price difference for my peaks is not huge, so tesla suggests  the battery sits full for backup if needed, and thats what I do.

I didnt get this setup to log until March of this year, so my data doesn't go that far back. I do have an interesting increase of capacity after a FW update though back in March.

I'm interested to see how this goes over time.

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/126296751/8c514666-67a8-45d6-8a32-0d1edc7daecc)

**Comment by MattMake** [comment]
Mine has only been measuring for a few months, max was 15.1, min 14.5.  I use my powerwalls every day/night with 20% reserve.

Granted, the specs for Powerwalls are 13.5kWh each, starting out at 15.x is exceeding published specs.  I wonder if the 70% capacity after 10 years is for the stated amount (13.5) or the actual amount (15.1)?  At 13.5, that means I should have 9.45kWh for each Powerwall in September, 2032.  I should achieve ROI on the solar+batteries in just about 9 years, so anything I get beyond that is just printed money :).

It will be nice to track this over time, thanks for taking the time to put this app together!

**Reply by jasonacox** [reply]
I do the same thing (use the Powerwall heavily each day/night).  I always wanted to add a "Warranty Trend" line to the PW capacity graph to show if my Powerwalls were within the estimated degradation. I never had time to get that done. If anyone figures it out, it would be a cool feature to add. :)

**Comment by jgleigh** [comment]
My four units started at 14 kWh. The Powerwall2 are down to 13.5 kWh and the Powerwall+ are down to 13.7 kWh after two years of use.

**Comment by youzer-name** [comment]
> I do the same thing (use the Powerwall heavily each day/night). I always wanted to add a "Warranty Trend" line to the PW capacity graph to show if my Powerwalls were within the estimated degradation. I never had time to get that done. If anyone figures it out, it would be a cool feature to add. :)

A horizonal warranty threshold line should be fairly easy to show in Grafana, but since you refer to it as a 'trend' line, I assume you mean a sloped line that goes towards the warranty level over the warranty time. but that would only be useful if the degradation is linear. 

If the PWs degrade to their warranty level in 12 months and then hold steady, they would be well below a linear degradation line but also still not subject to a warranty claim. I don't know how much of a parallel can be drawn between the car batteries and the PWs, but my understanding of the car battery degradation is that most of it occurs early, and then it levels off.

**Reply by MattMake** [reply]
I completely agree that the line may cross 'expected' ahead of time.  I personally would use it as a 'where am I' with battery degradation over time.  It wouldn't be of any practical use until closer to warranty expiration, but having the data over time would be instrumental in the event it is below the 70% threshold.  I'd have more ammunition going to Tesla with 10 years of data than a final kWh count near the 10 year mark.

**Comment by jasonacox** [comment]
My latest 2yr graphs show a gradual then clear increase in Powerwall energy capacity, after a year+ of degradation. I don't know if this is due to the recent firmware or the fact that I adjusted my system to go to grid power during the summer after peak (9pm) to handle a burst air conditioning load (our zone system activates to cools down the bedrooms at 9pm). In anycase, I'm happy to see this.

<img width="1026" alt="image" src="https://github.com/user-attachments/assets/d1e0b0eb-a38d-4c75-b47d-831e8d021521">

Example "on-grid for cooling" cycle:

<img width="898" alt="image" src="https://github.com/user-attachments/assets/abd00138-b253-49cf-a009-ab4a589e31a5">

**Comment by richardklam** [comment]
My April 2022 install was began logging on January 2023, and initially it showed 14.8 kWh, and then 2 months later bumped to 15.1 kWh (approximately 10 months old). I'm presently lat 14.2/14.3 kWh. What's interesting is the large capacity drops seem to be happening during the summer months when there is a lot AC usage. Most of the the year I only use 8-12 kWh during the peak, but summer it can be as much as 20 kWh. 

Apologies for the data gaps but that was during the Tesla API change. Of the measured energy there has been approximately 5.3 MWh charged, 4.6 MWh discharged (approximately 200 cycles per PW).

![image](https://github.com/user-attachments/assets/7ba66686-301b-4669-9941-be012a9d03ad)
